File: python/ccpayment/services/merchant_withdraw.py
"""Merchant Withdraw Service"""


class MerchantWithdrawService:
    def __init__(self, client):
        self.client = client

    def get_cwallet_user_id(self, cwallet_user_id: str):
        return self.client._post("/getCwalletUserId", {"cwalletUserId": cwallet_user_id})

    def get_withdraw_fee(self, coin_id: int, chain: str):
        return self.client._post("/getWithdrawFee", {"coinId": coin_id, "chain": chain})

    def apply_app_withdraw_to_network(self, data):
        return self.client._post("/applyAppWithdrawToNetwork", data)

    def apply_app_withdraw_to_cwallet(self, data):
        return self.client._post("/applyAppWithdrawToCwallet", data)

    def get_app_withdraw_record(self, data):
        return self.client._post("/getAppWithdrawRecord", data)

    def get_app_withdraw_record_list(self, data=None):
        return self.client._post("/getAppWithdrawRecordList", data or {})

    def get_auto_withdraw_record_list(self, data=None):
        return self.client._post("/getAutoWithdrawRecordList", data or {})

    def get_risk_refund_records(self, data=None):
        return self.client._post("/getRiskyRefundRecordList", data or {})

File: javascript/src/services/merchant-withdraw.js
class MerchantWithdrawService {
  constructor(client) { this.client = client; }
  async getCwalletUserId(cwalletUserId) { return this.client.post('/getCwalletUserId', { cwalletUserId }); }
  async getWithdrawFee(coinId, chain) { return this.client.post('/getWithdrawFee', { coinId, chain }); }
  async applyAppWithdrawToNetwork(data) { return this.client.post('/applyAppWithdrawToNetwork', data); }
  async applyAppWithdrawToCwallet(data) { return this.client.post('/applyAppWithdrawToCwallet', data); }
  async getAppWithdrawRecord(data) { return this.client.post('/getAppWithdrawRecord', data); }
  async getAppWithdrawRecordList(data = {}) { return this.client.post('/getAppWithdrawRecordList', data); }
  async getAutoWithdrawRecordList(data = {}) { return this.client.post('/getAutoWithdrawRecordList', data); }
  async getRiskRefundRecords(data = {}) { return this.client.post('/getRiskyRefundRecordList', data); }
}
module.exports = MerchantWithdrawService;
